python/cucim/src/cucim/skimage/measure/_regionprops_gpu.py

```diff
@@ -0,0 +1,882 @@
+        regionprops_feret_diameter_max(
+            out["image_convex"],
+            spacing=spacing,
+            props_dict=out,
+        )
+
+    compute_coords = "coords" in required_props
+    compute_coords_scaled = "coords_scaled" in required_props
+    if compute_coords or compute_coords_scaled:
+        regionprops_coords(
+            label_image,
+            max_label=max_label,
+            spacing=spacing,
+            compute_coords=compute_coords,
+            compute_coords_scaled=compute_coords_scaled,
+            props_dict=out,
+        )
+
+    if "label_filled" in required_props:
+        regionprops_label_filled(
+            label_image,
+            max_label=max_label,
+            props_dict=out,
+        )
+        if "area_filled" in required_props:
+            # also handles "num_pixels_filled"
+            out["area_filled"] = regionprops_area(
+                out["label_filled"],
+                max_label=max_label,
+                filled=True,
+                props_dict=out,
+            )
+        elif "num_pixels_filled" in required_props:
+            regionprops_num_pixels(
+                label_image,
+                max_label=max_label,
+                filled=True,
+                **perf_kwargs,
+                props_dict=out,
+            )
+        if "image_filled" in required_props:
+            out["image_filled"], _, _ = regionprops_image(
+                out["label_filled"],
+                max_label=max_label,
+                compute_image=True,
+                compute_convex=False,
+                props_dict=None,  # omit: using custom "image_filled" key
+            )
+
+    # If user had requested properties via their deprecated names, set the
+    # canonical names for the computed properties to the corresponding
+    # deprecated one.
+    for k, v in restore_legacy_names.items():
+        out[v] = out.pop(k)
+
+    # allow extra properties to be a list of callable functions
+    # logic is set to match that in regionprops/regionprops_table
+    for func in extra_properties:
+        if not isinstance(func, Callable):
+            raise ValueError(
+                "extra_properties must be a list of callable functions, got "
+                f"{type(func)}"
+            )
+        name = func.__name__
+        n_args = _infer_number_of_required_args(func)
+        images = out["image"]
+        # determine whether func requires intensity image
+        results = []
+        if n_args == 2:
+            if intensity_image is not None:
+                images_intensity = out["image_intensity"]
+                for image, image_intensity in zip(images, images_intensity):
+                    multichannel = label_image.shape < intensity_image.shape
+                    multichannel = label_image.shape < intensity_image.shape
+                    if multichannel:
+                        multichannel_list = [
+                            func(images, images_intensity[..., i])
```

Comment:
Just curious if `images` is intentional here, given that this statement is inside a for-loop using `image`.

Reply:
good catch! maybe there is no test case for this multi-channel intensity image case with a custom property. Let me test and push a fix

Reply:
Fixed as suggested and added a new test case to cover this code path